code/modules/surgery/organs/eyes.dm

```diff
@@ -126,10 +126,17 @@
 	name = "meson eyes"
 	desc = "These cybernetic eyes will allow you to see the structural layout of the station, and, well, everything else."
 	eye_color = "#199900"
-	vision_flags = SEE_TURFS
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
 	origin_tech = "materials=4;engineering=4;biotech=4;magnets=4"
 
+/obj/item/organ/internal/eyes/cybernetic/meson/insert(mob/living/carbon/human/M, special = FALSE)
+	ADD_TRAIT(M, TRAIT_MESON_VISION, "meson_vision")
```

Comment:
Because of the way traits work, it's possible someone could for example put on then take off meson goggles, or gain and lose the meson vision mutation, removing the trait given by the implant. This makes the source unique so it won't be removed in those other cases.

(Someone really needs to make a system to handle this automatically for all trait-granting items)
```suggestion
	ADD_TRAIT(M, TRAIT_MESON_VISION, "meson_vision[UID()]")
```

Reply:
This was one thing I was worried about but wasn't sure how to solve, thanks! I'll add this when I get home.